src/openapi/spec/api-request-schema.ts

```diff
@@ -0,0 +1,127 @@
+
+const schema = {
+    type: 'object',
+    required: ['features', 'version'],
+    properties: {
+        version: {
+            type: 'integer',
+            enum: [2],
+        },
+        features: {
+            type: 'array',
+            items: {
+                type: 'object',
+                required: ['name'],
+                additionalProperties: false,
+                properties: {
+                    name: {
+                        type: 'string',
+                    },
+                    description: {
+                        type: 'string',
+                    },
+                    enabled: {
+                        type: 'boolean',
+                    },
+                    stale: {
+                        type: 'boolean',
+                    },
+                    impressionData: {
+                        type: 'boolean',
+                    },
+                    strategies: {
+                        type: 'array',
+                        items: {
+                            type: 'object',
+                            required: ['name', 'constraints', 'parameters'],
+                            additionalProperties: false,
+                            properties: {
+                                name: {
+                                    type: 'string',
+                                },
+                                constraints: {
+                                    type: 'array',
+                                    items: {
+                                        type: 'object',
+                                        required: ['contextName', 'operator'],
+                                        properties: {
+                                            contextName: {
+                                                type: 'string',
+                                            },
+                                            operator: {
+                                                type: 'string',
+                                                enum: Object.values(Operator),
+                                            },
+                                            values: {
+                                                type: 'array',
+                                                items: {
+                                                    type: 'string',
+                                                },
+                                            },
+                                        },
+                                    },
+                                },
+                                parameters: {
+                                    type: 'object',
+                                },
+                            },
+                        },
+                    },
+                    variants: {
+                        type: 'array',
+                        items: {
+                            type: 'object',
```

Comment:
Could perhaps split this out as well:

https://github.com/Unleash/unleash/blob/main/src/lib/openapi/spec/variant-schema.ts